query-graphs-js/src/querygraph.ts

```diff
@@ -965,6 +954,22 @@ function federateSubgraphs(supergraph: Schema, subgraphs: QueryGraph[]): QueryGr
    
   }
 
+  // add contextual argument maps to builder
+  for (const [i, subgraph] of subgraphs.entries()) {
```

Comment:
previously this was calculated in a single loop above -> any reason to move it down here?

Reply:
Yeah, it's like that because we only need to do the calculations once at the end. The fact that it was inside the loop was causing naming colisions.